packages/contracts/contracts/SmartInvoiceEscrow.sol

```diff
@@ -94,21 +114,18 @@ contract SmartInvoiceEscrow is
     uint256 public milestone = 0;
     /// @notice Total amount released to provider so far
     uint256 public released = 0;
-    /// @notice Current dispute ID when using arbitrator resolution
-    uint256 public disputeId;
 
     /// @notice Latest MetaEvidence identifier for this escrow (ERC-1497)
     /// @dev Advances when off-chain details change (e.g., via _addMilestones),
     ///      and the current value is referenced by new Disputes
     uint256 public metaEvidenceId;
-    /// @notice Per-dispute Evidence group identifier (ERC-1497)
-    /// @dev Incremented *before* opening a new dispute so both Dispute and
-    ///      subsequent Evidence events for that dispute share the same group
-    uint256 public evidenceGroupId;
-
-    /// @notice Initializes the contract with wrapped ETH and factory addresses
-    /// @param _wrappedETH The address of the wrapped ETH contract
-    /// @param _factory The address of the SmartInvoiceFactory contract
+    /// @notice Arbitrator extra data for arbitrator resolvers (court and number of jurors)
+    bytes public arbitratorExtraData;
+    /// @notice Current dispute data when using arbitrator resolution
+    DisputeData private disputeData;
```

Comment:
I don't understand why we have to overwrite `disputeData` with each dispute. Why not store the data in array? This way we can use localDisputeID for `evidenceGroupID` instead of using 0 all the time. And also we can put `rounds` in dispute struct too.
Overwriting `disputeData` can lead to an issue with withdrawals which I will mention below

Reply:
I see, from the code it doesn't seem apparent that there is only one possible dispute. You can resolve it, `locked` flag gets dropped, you add more milestones and it looks like it can be locked again, no?